functions

```diff
@@ -79,11 +84,48 @@ service_create_container() {
   service_info "$SERVICE"
 }
 
+service_create_database() {
+  local SERVICE="$1"
+  local DATABASE="$2"
+  local SERVICE_ROOT="$PLUGIN_DATA_ROOT/$SERVICE"
+
+  [[ -n "$SERVICE" ]] || dokku_log_fail "Please specify a name for the service"
+  [[ -d "$SERVICE_ROOT" ]] || dokku_log_fail "$PLUGIN_SERVICE service $SERVICE does not exist"
+  [[ ! -e "$SERVICE_ROOT/databases/$DATABASE" ]] || dokku_log_fail "$PLUGIN_SERVICE service $SERVICE already has database $DATABASE"
+
+  local SERVICE_NAME="$(get_service_name "$SERVICE")"
+  local ROOTPASSWORD="$(cat "$SERVICE_ROOT/auth/root")"
+  local SERVICE_TTY_OPTS
+  has_tty && SERVICE_TTY_OPTS="-t"
+
+  local USERNAME=$(openssl rand -hex 8)
+  while [[ -e "$USERNAME" ]]; do
+    USERNAME=$(openssl rand -hex 8)
+  done
+  local PASSWORD=$(openssl rand -hex 8)
+
+  dokku_log_verbose_quiet "Creating user $USERNAME for service $SERVICE"
+  local CREATE_ROLE_SQL="CREATE USER \`$USERNAME\` IDENTIFIED BY '$PASSWORD';"
+  docker exec -i $SERVICE_TTY_OPTS "$SERVICE_NAME" env TERM="$TERM" mysql --user=root --password="$ROOTPASSWORD" --database="$SERVICE" -e "$CREATE_ROLE_SQL"
+  echo "$PASSWORD" > "$SERVICE_ROOT/auth/$USERNAME"
+  chmod 640 "$SERVICE_ROOT/auth/$USERNAME"
+
+  dokku_log_verbose_quiet "Creating database $DATABASE for service $SERVICE"
+  local CREATE_DB_SQL="CREATE DATABASE \`$DATABASE\`;"
```

Comment:
you can use single quotes and you won't have to do backtick escaping inside

Reply:
Do you mean using single quotes instead of backticks?

Reply:
no, instead of double quotes

Reply:
But i need to interpolate `$DATABASE` into the string.

Reply:
yea.. but in this case you don't have to put double quotes there anyway:
```sql
'CREATE DATABASE `'$DATABASE'`;'
```

Reply:
This is why I often think of (ab)using [Bash Infinity](/niieani/bash-oo-framework) :D